File: notification_hub_backend_service/src/flutter/push_demo/android/app/src/main/java/com/schauer/pushdemo/services/NotificationRegistrationService.java
package com.schauer.pushdemo.services;

import io.flutter.embedding.engine.FlutterEngine;
import io.flutter.plugin.common.MethodChannel;

public class NotificationRegistrationService {

    public static final String NOTIFICATION_REGISTRATION_CHANNEL = "com.schauer.pushdemo/notificationregistration";
    public static final String REFRESH_REGISTRATION = "refreshRegistration";

    private MethodChannel notificationRegistrationChannel;

    public NotificationRegistrationService(FlutterEngine flutterEngine) {
        notificationRegistrationChannel = new MethodChannel(flutterEngine.getDartExecutor().getBinaryMessenger(), NOTIFICATION_REGISTRATION_CHANNEL);
    }

    public void refreshRegistration() {
        notificationRegistrationChannel.invokeMethod(REFRESH_REGISTRATION, null);
    }
}

File: notification_hub_backend_service/src/flutter/push_demo/android/app/src/main/kotlin/com/schauer/pushdemo/services/NotificationRegistrationService.kt
package com.schauer.pushdemo.services

import io.flutter.embedding.engine.FlutterEngine
import io.flutter.plugin.common.MethodChannel

class NotificationRegistrationService {

    companion object {
        const val NOTIFICATION_REGISTRATION_CHANNEL = "com.schauer.pushdemo/notificationregistration"
        const val REFRESH_REGISTRATION = "refreshRegistration"
    }

    private var notificationRegistrationChannel : MethodChannel

    constructor(flutterEngine: FlutterEngine) {
        notificationRegistrationChannel = MethodChannel(flutterEngine.dartExecutor.binaryMessenger, NOTIFICATION_REGISTRATION_CHANNEL)
    }

    fun refreshRegistration() {
        notificationRegistrationChannel.invokeMethod(REFRESH_REGISTRATION, null)
    }
}